plugins/net_plugin/net_plugin.cpp

```diff
@@ -864,15 +863,15 @@ namespace eosio {
        *  @{
        */
       // See NTP protocol. https://datatracker.ietf.org/doc/rfc5905/
-      tstamp                         org{0}; //!< origin timestamp. Time at the client when the request departed for the server.
-      // tstamp (not used)           rec{0}; //!< receive timestamp. Time at the server when the request arrived from the client.
-      tstamp                         xmt{0}; //!< transmit timestamp, Time at the server when the response left for the client.
-      // tstamp (not used)           dst{0}; //!< destination timestamp, Time at the client when the reply arrived from the server.
+      std::chrono::nanoseconds               org{0}; //!< origin timestamp. Time at the client when the request departed for the server.
+      // std::chrono::nanoseconds (not used) rec{0}; //!< receive timestamp. Time at the server when the request arrived from the client.
+      std::chrono::nanoseconds               xmt{0}; //!< transmit timestamp, Time at the server when the response left for the client.
+      // std::chrono::nanoseconds (not used) dst{0}; //!< destination timestamp, Time at the client when the reply arrived from the server.
       /** @} */
       // timestamp for the lastest message
-      tstamp                         latest_msg_time{0};
-      tstamp                         hb_timeout{std::chrono::milliseconds{def_keepalive_interval}.count()};
-      tstamp                         latest_blk_time{0};
+      std::chrono::system_clock::time_point       latest_msg_time{std::chrono::system_clock::time_point::min()};
+      std::chrono::milliseconds                   hb_timeout{std::chrono::milliseconds{def_keepalive_interval}};
+      std::chrono::system_clock::time_point       latest_blk_time{std::chrono::system_clock::time_point::min()};
```

Comment:
I'd prefer have these all be `tstamp` as before, but with `tstamp` defined as `std::chrono::nanoseconds`. 

If we want to differentiate between a point in time (since the epoch), i.e `time_point`, and a `duration`, then org/rec/xmt/dst should be `time_point`, and not durations.

Also I don't think it is a good idea to use `std::chrono::system_clock::time_point` ourselves because the duration of the system_clock is not specified. 

So I think the most straightforward thing might be to use `tstamp` defined as `std::chrono::nanoseconds` for both actual timestamps (i.e. time since epoch) and durations. Because it is a duration from std::chrono, we benefit from type safety insuring that we don't add `std::chrono::nanoseconds` and `std::chrono::milliseconds`, but we use a consistent internal representation. I think it would make the code simpler and clearer.

Reply:
I created #1326 as this is a merge up PR, I think it makes sense to make these changes in a separate PR.